### PR TITLE
feat: avoid OutputAsset#name clone asset

### PR DESCRIPTION
**Pull request by shulaoda**
### Description
Related to #1586 

https://github.com/rolldown/rolldown/blob/73808a9dfce628a7083b1aad6a80b63ad0b013ed/crates/rolldown_common/src/file_emitter.rs#L86-L91

### CODE REVIEW

**Comment by shulaoda** [comment]
@underfin cc

**Comment by IWANABETHATGUY** [comment]
/Users/runner/work/_temp/3c2fef40-cddd-4593-8d5d-7a047a72e37e.sh: line 1:  4534 Segmentation fault: 11  pnpm run --recursive --parallel --filter=!rollup-tests test

File: crates/rolldown_binding/src/types/binding_outputs.rs [hunk 1]
```diff
@@ -96,7 +99,10 @@ impl FinalBindingOutputs {
       let mut inner = inner.lock().expect("PoisonError raised");
       inner.iter_mut().for_each(|o| match o {
         rolldown_common::Output::Asset(asset) => {
-          assets.push(BindingOutputAsset::new(unsafe { std::mem::transmute(asset.as_mut()) }));
+          assets.push(BindingOutputAsset::new(unsafe {
+            &mut *(std::ptr::from_mut::<std::sync::Arc<rolldown_common::EmittedAsset>>(asset)
+              .cast::<rolldown_common::EmittedAsset>())
+          }));
```
Comment (IWANABETHATGUY) [inline crates/rolldown_binding/src/types/binding_outputs.rs]: I don't thnk this is a safe cast.

Comment (shulaoda) [reply crates/rolldown_binding/src/types/binding_outputs.rs]: Yes, I am currently investigating this problem. I'm not sure if we should really avoid cloning, it seems to affect the `files` in `FileEmitter`.

Comment (IWANABETHATGUY) [reply crates/rolldown_binding/src/types/binding_outputs.rs]: According to the benchmark `asset` clone is not the bottleneck, clone it is fine.

**Comment by shulaoda** [comment]
When `emitted_files` exist, it seems that the assets in `FileEmitter`'s `files` are no longer in use. Perhaps we can delete it directly.

**Comment by shulaoda** [comment]
It seems necessary, I will close this PR.

**Pull request closed without merging**